presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/SqlServerQueryRunner.java

```diff
@@ -0,0 +1,101 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.sqlserver;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Streams;
+import io.airlift.tpch.TpchTable;
+import io.prestosql.Session;
+import io.prestosql.metadata.QualifiedObjectName;
+import io.prestosql.plugin.tpch.TpchPlugin;
+import io.prestosql.spi.security.Identity;
+import io.prestosql.testing.QueryRunner;
+import io.prestosql.tests.DistributedQueryRunner;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.prestosql.plugin.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static io.prestosql.tests.QueryAssertions.copyTable;
+import static java.util.Locale.ENGLISH;
+
+public final class SqlServerQueryRunner
+{
+    private SqlServerQueryRunner() {}
+
+    public static final String CATALOG = "sqlserver";
+
+    private static final String TEST_SCHEMA = "dbo";
+
+    public static QueryRunner createSqlServerQueryRunner(TpchTable<?>... tables)
+            throws Exception
+    {
+        return createSqlServerQueryRunner(ImmutableMap.of(), ImmutableList.copyOf(tables));
+    }
+
+    public static QueryRunner createSqlServerQueryRunner(Map<String, String> connectorProperties, Iterable<TpchTable<?>> tables)
+            throws Exception
+    {
+        TestingSqlServerServer.waitForSqlServer();
+
+        DistributedQueryRunner queryRunner = new DistributedQueryRunner(createSession(), 3);
+        try {
+            queryRunner.installPlugin(new TpchPlugin());
+            queryRunner.createCatalog("tpch", "tpch");
+
+            connectorProperties = new HashMap<>(ImmutableMap.copyOf(connectorProperties));
+            connectorProperties.putIfAbsent("connection-url", TestingSqlServerServer.getJdbcUrl());
+            connectorProperties.putIfAbsent("connection-user", TestingSqlServerServer.USER);
+            connectorProperties.putIfAbsent("connection-password", TestingSqlServerServer.PASSWORD);
+            connectorProperties.putIfAbsent("allow-drop-table", "true");
+
+            queryRunner.installPlugin(new SqlServerPlugin());
+            queryRunner.createCatalog(CATALOG, "sqlserver", connectorProperties);
+
+            provisionTables(createSession(), queryRunner, tables);
+
+            return queryRunner;
+        }
+        catch (Throwable e) {
+            closeAllSuppress(e, queryRunner);
+            throw e;
+        }
+    }
+
+    private static synchronized void provisionTables(Session session, QueryRunner queryRunner, Iterable<TpchTable<?>> tables)
```

Comment:
Make sure this is no longer `static synchronized` when you move container init to be once-per-test-class.